_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,265 @@
+
+![npm trends screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/01.png)
+###### Source [npm trends](https://www.npmtrends.com/@testing-library/react-vs-enzyme)
+
+## EnzymeJS
+From the enzyme documentation we can read a short description: *Enzyme is a JavaScript Testing utility for React that makes it easier to test your React Components' output. You can also manipulate, traverse, and in some ways simulate runtime given the output.*
+
+Enzyme was released in **2015 by AirBnB**. And let's face it, Enzyme has gained a lot of popularity. Last year, the library was moved from AirBnB space to external github space as a separate independent organization. This step was intended to encourage the community to further develop the library.
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/02.png)
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/03.png)
+###### Source [Enzyme github](https://github.com/enzymejs/enzyme/commit/b06750bb5f6248ceb9c3fae903a71d0747a420d6)
+
+When it comes to raw data, the latest version of **Enzyme 3.11.0** has a bundle size at **463.2kb (127.5 kB)** and was released on **20 December 2019**. Currently has 258 [open issues](https://github.com/enzymejs/enzyme/issues) and 31 open PR’s. 	
+
+![bundlephobia enzyme sreen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/04.png)
+###### Source [Bundle Phobia](https://bundlephobia.com/package/enzyme@3.11.0)
+
+
+Interesting fact: 
+- There are also Enzyme Adapter Packages for React. For example, for version 16, it also needs *enzyme-adapter-react-16* to be installed. Unfortunately, it does not support all features completely.
+
+## React Testing Library (RTL)
+
+RTL says about itself: *The React Testing Library is a very light-weight solution for testing React components. It provides light utility functions on top of react-dom and react-dom/test-utils, in a way that encourages better testing practices.*.
+
+Released in **April 2018 by Kent C. Dodds**. The latest stable version **12.1.2** of RTL has a bundle size at **215.8kb (47.3 kB)** and was released in **October 2021**. Currently, a pre-release of v13.0.0-alpha is being prepared. Repository has 26 [open issues](https://github.com/enzymejs/enzyme/issues) and 2 open PR’s.  
+
+![bundlephobia rtl screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/05.png)
+
+###### Source [Bundle Phobia](https://bundlephobia.com/package/@testing-library/react@12.1.2)
+
+
+Interesting facts: 
+- When you create a project by Create React App, you will get support for React Testing Library right away. 
```

Comment:
also React is officially recommending RTL as testing library https://reactjs.org/docs/testing.html#tools